### ISSUE TITLE
DI for inherited class

### ISSUE THREAD

**Issue by klinki** [angular/angular#7627]
**Note: for support questions, please use one of these channels:** https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question. This repository's issues are reserved for feature requests and bug reports.

* **I'm submitting a ... **
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, see note at the top of this template.

* **Do you want to request a *feature* or report a *bug*?**

Report a bug

* **What is the current behavior?**

Hi, I just found a bug when TypeScript inheritance is used.

Scenario:

```
class A {
    constructor(lots of dependencies here....) {}
}


@Injectable()
class B  extends A {}
```

When class B does not have a constructor, dependencies are not properly injected.

But when you provide a constructor which calls the parent constructor:

```
@Injectable()
class B  extends A {
 constructor(lots of dependencies here....) {
    super(lots of dependencies here...);
  }
}
```
everything works fine. 

* **Please tell us about your environment:**

- Angular version: 2.0.0-beta.9
- Browser: [Chrome 48 ]
- Language: [TypeScript 1.8]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

**Comment by pkozlowski-opensource** [comment]
What you are requesting here is that TS auto-generates a constructor in the class B. This was discussed before and would be more a request to language authors than angular. Closing, as this is not something angular can "solve".

**Comment by klinki** [comment]
But if I instanciate class by using new operator, it works.

`
var instanceOfB = new B(lots of dependencies manually passed here....); 
`

instanceOfB will be properly created with all dependencies.

**Comment by tbosch** [comment]
@pkozlowski-opensource we do support inheriting metadata from the parent class if the ctor is not overwritten. Typescript inherits that parent ctor by default as well.

Here are the details of our semantics: https://github.com/angular/angular/commit/f5c8e0989d85bc064f689fc3595207dfb29413f4

But we aware that there is an open bug when using in JIT mode and with Closure Compiler: https://github.com/angular/angular/issues/13211

**Comment by tbosch** [comment]
Note that this was fixed in Angular 2.3.x, i.e. does not work in 2.0.0-beta.9
